src/main/java/mops/businesslogic/query/FileQueryBuilder.java

```diff
@@ -0,0 +1,262 @@
+
+            if (OWNERS.contains(token) && index + 1 < length) {
+                owner(tokens.get(++index));
+            } else if (NAMES.contains(token) && index + 1 < length) {
+                name(tokens.get(++index));
+            } else if (TYPES.contains(token) && index + 1 < length) {
+                type(tokens.get(++index));
+            } else if (TAGS.contains(token) && index + 1 < length) {
+                tag(tokens.get(++index));
+            } else { // values without keys are file names
+                name(oldToken);
+            }
+
+            index++;
+        }
+
+        return this;
+    }
+
+    // pmd doesn't like tokenizers
+    @SuppressWarnings({ "PMD.CyclomaticComplexity", "PMD.NPathComplexity", "PMD.DataflowAnomalyAnalysis" })
+    private List<String> tokenize(String search) {
+        int length = search.length();
+
+        boolean isToken = false;
+        boolean isString = false;
+        StringBuilder tokenBuilder = new StringBuilder();
+        List<String> tokens = new ArrayList<>();
+
+        for (int index = 0; index < length; index++) {
+            char current = search.charAt(index);
+
+            if (!isToken && Character.isWhitespace(current)) {
+                continue;
+            }
+
+            if (isToken && !isString && (Character.isWhitespace(current) || current == KEY_SEPARATOR)) {
+                isToken = false;
+                String token = tokenBuilder.toString();
+                tokenBuilder.setLength(0);
+                tokens.add(token);
+                continue;
+            }
+
+            if (current == STRING_MARKER) {
+                if (!isToken) {
+                    isToken = true;
+                    isString = true;
+                    continue;
+                }
+                if (isString) {
+                    isToken = false;
+                    isString = false;
+                    String token = tokenBuilder.toString();
+                    tokenBuilder.setLength(0);
+                    tokens.add(token);
+                    continue;
+                }
+            }
+
+            if (!isToken) {
+                isToken = true;
+            }
+
+            tokenBuilder.append(current);
+        }
+
+        if (isToken) {
+            String token = tokenBuilder.toString();
+            tokenBuilder.setLength(0);
+            tokens.add(token);
+        }
+
+        return List.copyOf(tokens);
+    }
+
+    /**
+     * Add all owners.
+     *
+     * @param owners list of owner to search for
+     * @return this
+     */
+    public FileQueryBuilder owners(@NonNull Iterable<String> owners) {
+        owners.forEach(this::owner);
+        return this;
+    }
+
+    /**
+     * Add an owner.
+     *
+     * @param owner one Owner
+     * @return this
+     */
+    public FileQueryBuilder owner(@NonNull String owner) {
+        if (owner.isEmpty()) {
+            throw new IllegalArgumentException("owner must not be empty!");
```

Comment:
Can you add a logging line here?

Reply:
no, you are doing this in your pr.

Reply:
There shouldn't be logging necessary, it's internal logic